GitCommands/Settings/ConfigFileSettings.cs

```diff
@@ -70,15 +70,28 @@ public static ConfigFileSettings CreateGlobal(ConfigFileSettings? lowerPriority,
                 SettingLevel.SystemWide);
         }
 
-        public new string GetValue(string setting)
-        {
-            return GetString(setting, string.Empty);
-        }
+        public new string GetValue(string setting) => GetString(setting, string.Empty);
 
-        public IReadOnlyList<string> GetValues(string setting)
-        {
-            return SettingsCache.GetValues(setting);
-        }
+        /// <summary>
+        /// Get a git setting converted in an expected C# value type (bool, int, ...)
+        /// </summary>
+        /// <typeparam name="T">the expected type to convert the value to.</typeparam>
+        /// <param name="setting">the git setting key</param>
+        /// <returns>
+        /// null if the settings is not set
+        /// the value converted in the <typeparamref name="T" /> type otherwise.
+        /// </returns>
+        public T? GetValue<T>(string setting) where T : struct => ConvertValue<T>(GetValue(setting));
+
+        private T? ConvertValue<T>(string value) where T : struct
+            => string.IsNullOrWhiteSpace(value) ? default : (T)Convert.ChangeType(value, typeof(T));
```

Comment:
https://github.com/gitextensions/gitextensions/pull/11335#discussion_r1405421710

Same feedback. Need to have a git config defaults functionality from git config documentation. Not every git config default equals default of .net type.

Reply:
No, with this Api, we really want to know if the setting has been set or not, so we don't want to have a default value because we won't know anymore if the setting has been set. And we need this information so that the rebase command line arguments are built differently.

Reply:
This function returns `T?`. So `default` means `null`. Why not write it as such?

Reply:
The only good reason is that I initially put null but I was asked to use default instead (but it was maybe part of Api discussion). I prefer null also.

Reply:
Me 3.

Reply:
Also set a config value to truuuueee and see if you can crash GE?

Reply:
Yes, you are right, @vbjay. This resulting exception needs to be wrapped into a `UserExternalOperationException`.

Reply:
> This resulting exception needs to be wrapped into a `UserExternalOperationException`.

After investigation, it seems to me difficulty doable (or at least a lot of work) and maybe a little hacky:
* the `UserExternalOperationException` exception is not available here
* `UserExternalOperationException` is not really made to handle something else than an external error
* where the settings are read, in `OnShow()`, the exception is unwrapped by the `Control.cs` code to keep only the inner exception and so even if a `UserExternalOperationException` is thrown, that not type of the one received and so the special behavior to display a more user friendly popup can't work. We still can move some code to not do things in  `OnShow()` but not sure everywhere like that.
* the popup seems weird because that's not really an external command that cause the issue.

To work around all these issues, maybe the easier is to throw a new type of exception like `GitConfigFormatException` and the resulting error popup will be  (text could be customized):
![image](https://github.com/gitextensions/gitextensions/assets/460196/9eced76f-8338-4063-9ed2-9a9f2c4c47e2)